### PR TITLE
Enable gocritic and gocyclo linters

### LINKED ISSUE #2248: Enable gocritic and gocyclo linters
**Is your feature request related to a problem? Please describe.**
Enable `gocritic` and `gocyclo` linters to improve overall code quality.

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

- Enable `gocritic` and `gocyclo` linters.
- Fix issues detected

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #2248 

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by viveksinghggits** [comment]
Let's mention the linter names in the PR title as well.

**Review by viveksinghggits — APPROVED** [review]

File: pkg/kanctl/actionset.go [hunk 1]
```diff
@@ -503,7 +501,7 @@ func parseGenericObjectReference(s string) (crv1alpha1.ObjectReference, error) {
 	}, nil
 }
 
-func parseObjectsFromSelector(selector, kind, sns string, cli kubernetes.Interface, osCli osversioned.Interface, parsed map[string]bool) ([]crv1alpha1.ObjectReference, error) {
+func parseObjectsFromSelector(selector, kind, sns string, cli kubernetes.Interface, osCli osversioned.Interface, parsed map[string]bool) ([]crv1alpha1.ObjectReference, error) { //nolint:gocyclo
```
Comment (pavannd1) [inline pkg/kanctl/actionset.go:504]: Does this fail even with 30 cyclo limit?

File: pkg/kube/volume/volume.go [hunk 2]
```diff
@@ -309,6 +309,6 @@ func labelSelector(labels map[string]string) string {
 
 // zoneToRegion removes -latter or just last latter from provided zone.
 func zoneToRegion(zone string) string {
-	r, _ := regexp.Compile("-?[a-z]$")
+	r, _ := regexp.Compile("-?[a-z]$") //nolint:gocritic
```
Comment (pavannd1) [inline pkg/kube/volume/volume.go:312]: We should add a TODO or some comment to say why we are disabling the linter

**Review by pavannd1 — CHANGES_REQUESTED** [review]
Will re-review once we address the comments

File: pkg/kanctl/actionset.go (hunk 1 above)
Comment (ankitjain235) [reply pkg/kanctl/actionset.go:504]: We are setting the limit to 20, that's where it's failing.

**Comment by ankitjain235** [comment]
@pavannd1 PTAL at the PR again.

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**